src/main/java/qupath/ext/imglib2/ImgBuilder.java

```diff
@@ -205,14 +219,203 @@ private ImgBuilder(ImageServer<BufferedImage> server, T type, Function<BufferedI
+    }
+
+    /**
+     * Create a builder from an {@link ImageServer}. This doesn't create any accessibles yet.
+     * <p>
+     * The provided type must be compatible with the input image:
+     * <ul>
+     *     <li>
+     *         If the input image is {@link ImageServer#isRGB() RGB}, the type must be {@link UnsignedByteType}. Images
+     *         created by the returned builder will have 4 channels (RGBA).
+     *     </li>
+     *     <li>
+     *         Else:
+     *         <ul>
+     *             <li>
+     *                 If the input image has the {@link PixelType#UINT8} {@link ImageServer#getPixelType() pixel type},
+     *                 the type must be {@link UnsignedByteType}.
+     *             </li>
+     *             <li>
+     *                 If the input image has the {@link PixelType#INT8} {@link ImageServer#getPixelType() pixel type},
+     *                 the type must be {@link ByteType}.
+     *             </li>
+     *             <li>
+     *                 If the input image has the {@link PixelType#UINT16} {@link ImageServer#getPixelType() pixel type},
+     *                 the type must be {@link UnsignedShortType}.
+     *             </li>
+     *             <li>
+     *                 If the input image has the {@link PixelType#INT16} {@link ImageServer#getPixelType() pixel type},
+     *                 the type must be {@link ShortType}.
+     *             </li>
+     *             <li>
+     *                 If the input image has the {@link PixelType#UINT32} {@link ImageServer#getPixelType() pixel type},
+     *                 the type must be {@link UnsignedIntType}.
+     *             </li>
+     *             <li>
+     *                 If the input image has the {@link PixelType#INT32} {@link ImageServer#getPixelType() pixel type},
+     *                 the type must be {@link IntType}.
+     *             </li>
+     *             <li>
+     *                 If the input image has the {@link PixelType#FLOAT32} {@link ImageServer#getPixelType() pixel type},
+     *                 the type must be {@link FloatType}.
+     *             </li>
+     *             <li>
+     *                 If the input image has the {@link PixelType#FLOAT64} {@link ImageServer#getPixelType() pixel type},
+     *                 the type must be {@link DoubleType}.
+     *             </li>
+     *         </ul>
+     *     </li>
+     * </ul>
+     *
+     * @param server the input image
+     * @param type the expected type of the output image
+     * @return a builder to create an instance of this class
+     * @param <T> the type corresponding to the provided image
+     * @throws IllegalArgumentException if the provided type is not compatible with the input image (see above), or if
+     * the provided image has less than one channel
+     */
+    public static <T extends NativeType<T> & RealType<T>> ImgBuilder<T, ?> createRealBuilder(ImageServer<BufferedImage> server, T type) {
```

Comment:
I find the order `<T extends RealType <T> & NativeType<T>>` more logical - the important thing is the `RealType`, while `NativeType` is more of an implementation detail (albeit necessary to know about sometimes).